crates/allocator/src/bump.rs

```diff
@@ -194,6 +195,11 @@ impl InnerAlloc {
             Some(alloc_start)
         }
     }
+
+    /// Aligns the start pointer of the next allocation.
+    fn align_ptr(&self, layout: &Layout) -> usize {
+        (self.next + layout.align() - 1) & !(layout.align() - 1)
```

Comment:
Please add a comment with the reasoning behind the calculation and the `AND` here.

Reply:
maybe some concrete example will help here